x-pack/elastic-agent/pkg/core/plugin/process/app.go

```diff
@@ -192,16 +196,32 @@ func (a *Application) watch(ctx context.Context, p app.Taggable, proc *process.I
 		case ps := <-a.waitProc(proc.Process):
 			procState = ps
 		case <-a.bgContext.Done():
-			a.Stop()
+			return
+		case <-ctx.Done():
+			// closer called
 			return
 		}
 
 		a.appLock.Lock()
 		if a.state.ProcessInfo != proc {
+			// kill original process if possible
+			if proc != nil && proc.Process != nil {
+				_ = proc.Process.Kill()
```

Comment:
shall we wait or check for other errors?

Reply:
this is best effort